### ISSUE TITLE
NestedBlockDepth and functions like 'use'

### ISSUE THREAD

**Issue by sanyarnd** [detekt/detekt#5030]
Imagine the following code:
```kotlin
closeable1.use { closeable1 ->
    closeable2.use { closeable2 ->
        closeable3.use { closeable3 ->
            repeat(N) {
                // ...logic...
            }
        }
    }
}
```

`NestedBlockDepth` is naturally violated here, but there's no way to rewrite this code.
The only way to bypass it is suppression (and I don't like it) or setting threshold higher (it is equivalent to disabling the check).

I think this check needs something `ComplexMethod` has, like `ignoreNestingFunctions`.

**Comment by schalkms** [comment]
Why not refactor this nested code into well-named functions to make it easier to understand? What's preventing you to do this?

By the way, the config option `ComplexMethod:nestingFunctions` actually counts the formalized McCabe complexity for scope functions. This is completely unrelated to the `NestedBlcokDepth` rule.

Do you know that you could write a [custom detekt rule](https://detekt.dev/docs/introduction/extensions#custom-rulesets) exactly for your scenario?

**Comment by sanyarnd** [comment]
The code above is equivalent to
```java
try (var r1 = SomeStream; 
     var r2 = OtherStream; 
     var r3 = AnotherStream) {
}
```

I don't think it needs to be rewritten, it's just a lack of such syntax in Kotlin that produces the nested `use` calls.

**Comment by cortinico** [comment]
> NestedBlockDepth is naturally violated here, but there's no way to rewrite this code.

I agree with @sanyarnd I had to write this code a couple of times (It was anyway never more than 4/5 nested blocks, so that would be a good threshold imho).

We could look into ignoring blocks where the receiver is a `Closeable` to handle those scenarios.

**Comment by BraisGabin** [comment]
Are those so many cases? An `@Suppress` wouldn't it be enough? You can always add a new function to "reset" the indentation. For me it is a bit strange to make `NestedBlockDepth` more complex. This open the door to add even more exceptions to that rule.
